Solutions/ThreatXCloud/Playbooks/ThreatXPlaybooks/ThreatX-Enrichment/azuredeploy.json

```diff
@@ -0,0 +1,707 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "metadata": {
+        "title": "Fetch Threat Intel from ThreatX",
+        "description": "This playbook provides/updates the threat intel and essential details in comments section of triggered incident so that SOC analysts can directly take corrective measure to stop the attack",
+        "prerequisites": [
+            "1. ThreatX-WAF Custom Connector needs to be deployed prior to the deployment of this playbook under the same subscription.",
+            "2. API Key . To get API Key, login into your ThreatX cloud instance dashboard and navigate to Settings --> API Key --> Add Api key"
+        ],
+        "postDeployment": [
+            "1. During deployment you need to provide your api key with key vault name and tenant name."
+        ],
+        "prerequisitesDeployTemplateFile": "../../ThreatXCustomConnector/azuredeploy.json",
+        "lastUpdateTime": "",
+        "entities": [ "ip" ],
+        "tags": [ "Enrichment", "Notification", "ThreatIntelligence", "ThreatX", "WAF" ],
+        "support": {
+            "tier": "microsoft",
+            "armtemplate": "Generated from https://github.com/Azure/Azure-Sentinel/tree/master/Tools/Playbook-ARM-Template-Generator"
+        },
+        "author": {
+            "name": "microsoft"
+        }
+    },
+    "parameters": {
+        "PlaybookName": {
+            "defaultValue": "ThreatX-Enrichment",
+            "type": "string"
+        },
+        "customer_name": {
+            "type": "string",
+            "metadata": {
+                "description": "Enter value for Tenant name"
+            }
+        },
+        "Keyvault name": {
+            "defaultValue": "",
+            "type": "String",
+            "metadata": {
+                "description": "Enter unique name else deployment will fail: postfix with some random number"
+            }
+        },
+        "Threatx Key name": {
+            "defaultValue": "",
+            "type": "string",
+            "metadata": {
+                "description": "Your Key name"
+            }
+        },
+        "Threatx Key": {
+            "defaultValue": "",
+            "type": "securestring",
+            "metadata": {
+                "description": "Your secure threat-x key."
+            }
+        },
+        "CustomConnectorName": {
+            "defaultValue": "ThreatX-WAFCustomConnector",
+            "type": "string",
+            "metadata": {
+                "description": "Name of the custom connector, if you want to change the default name, make sure to use the same in all ThreatXCloud automation playbooks as well"
+            }
+        }
+    },
+    "variables": {
+        "keyvault_name": "[toLower(parameters('Keyvault name'))]",
```

Comment:
Same here. Keyvault name should be globally unique, and user may choose a name which is not unique.

Reply:
Rahul actually this has been left as it is because what if customer wants to use here existing key vault for storing keys and secrets and in metadata I have mentioned to please pick up the unique name.